datadog_checks_downloader/datadog_checks/downloader/download.py

```diff
@@ -86,52 +79,41 @@ def __init__(
         # NOTE: This updater will store files under:
         # os.path.join(REPOSITORIES_DIR, REPOSITORY_DIR)
         self.__updater = Updater(
-            REPOSITORY_DIR,
-            {
-                'repo': {
-                    'url_prefix': repository_url_prefix,
-                    'metadata_path': 'metadata.staged',
-                    'targets_path': 'targets',
-                    'confined_target_dirs': [''],
-                }
-            },
+            metadata_dir=os.path.join(REPOSITORIES_DIR, REPOSITORY_DIR, 'metadata/current'),
+            metadata_base_url=f'{repository_url_prefix}/metadata.staged/',
+            target_base_url=f'{repository_url_prefix}/targets/',
+            target_dir=self.__targets_dir,
         )
 
+        # Increase requests timeout.
+        # There's no officially supported way to do this without either writing our own
+        # fetcher from scratch or relying on internals. We're choosing the latter for now.
+        # - https://github.com/theupdateframework/python-tuf/blob/v2.0.0/tuf/ngclient/updater.py#L99
+        # - https://github.com/theupdateframework/python-tuf/blob/v2.0.0/tuf/ngclient/_internal/requests_fetcher.py#L49
+        self.__updater._fetcher.socket_timeout = 60
+
         # NOTE: Update to the latest top-level role metadata only ONCE, so that
         # we use the same consistent snapshot to download targets.
         self.__updater.refresh()
 
     def __download_with_tuf(self, target_relpath):
-        target = self.__updater.get_one_valid_targetinfo(target_relpath)
-        updated_targets = self.__updater.updated_targets((target,), self.__targets_dir)
+        target = self.__updater.get_targetinfo(target_relpath)
+        target_abspath = os.path.join(self.__targets_dir, target_relpath)
+        local_relpath = self.__updater.find_cached_target(target, target_abspath)
 
         # Either the target has not been updated...
-        if not len(updated_targets):
+        if local_relpath:
             logger.debug('%s has not been updated', target_relpath)
-        # or, it has been updated, in which case...
+        # or, it has been updated, in which case we download the new version
         else:
-            # First, we use TUF to download and verify the target.
-            if len(updated_targets) != 1:
-                raise UpdatedTargetsError(
-                    'Expecting only one target {!r} to be updated; got: {}'.format(target, ', '.join(updated_targets))
-                )
-
-            updated_target = updated_targets[0]
-
-            if updated_target != target:
-                raise UpdatedTargetsError(
-                    'Unknown target updated, expected {!r} but got {!r}'.format(target, updated_target)
-                )
-
-            self.__updater.download_target(updated_target, self.__targets_dir)
+            os.makedirs(os.path.dirname(target_abspath), exist_ok=True)
```

Comment:
Why do we need to make the parent dirs now (if need be)? Presumably because the client no longer automatically makes it for you? If so, that's a bit of an odd design decision on the part of the client, I'd imagine.

Reply:
Old client created them for you: https://github.com/theupdateframework/python-tuf/blob/e86a61571c332643831a56d1c2910f2475f08998/tuf/client/updater.py#L3056-L3064

New client doesn't: https://github.com/theupdateframework/python-tuf/blob/7ada2af384442f1c3775a4bcbd9f33270c1c8fa0/tuf/ngclient/updater.py#L199